traffic_portal/app/src/common/api/ServerCapabilityService.js

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+var ServerCapabilityService = function($http, ENV, locationUtils, messageModel) {
+
+	this.getServerCapabilities = function(queryParams) {
+		return $http.get(ENV.api['root'] + 'server_capabilities', {params: queryParams}).then(
+			function(result) {
+				return result.data.response;
+			},
+			function(err) {
+				throw err;
+			}
+		);
+	};
+
+	this.getServerCapability = function(name) {
+		return $http.get(ENV.api['root'] + 'server_capabilities?name=' + name).then(
```

Comment:
instead of string concatenation, use the `params` key of the optional args object, like you do above in `getServerCapabilty`. This ensures proper URI Encoding.

```javascript
return $http.get(ENV.api['root'] + 'server_capabilities', {params: {"name": name}}).then(r=>r.data.response[0],e=>{throw e;})
```